File: app/Models/ExpenseCategory.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Support\Str;

class ExpenseCategory extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'uuid',
        'user_id',
        'name',
        'icon',
        'color',
    ];

    protected function casts(): array
    {
        return [
            'deleted_at' => 'datetime',
        ];
    }

    protected static function booted()
    {
        static::creating(function ($category) {
            if (empty($category->uuid)) {
                $category->uuid = (string) Str::uuid();
            }
        });
    }

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function expenses()
    {
        return $this->hasMany(Expense::class);
    }
}

File: flutter_expense/lib/models/expense_category.dart
class ExpenseCategory {
  final int id;
  final String uuid;
  final String name;
  final String? icon;
  final String? color;

  const ExpenseCategory({
    required this.id,
    required this.uuid,
    required this.name,
    this.icon,
    this.color,
  });

  factory ExpenseCategory.fromJson(Map<String, dynamic> json) {
    return ExpenseCategory(
      id: json['id'] as int,
      uuid: json['uuid'] as String? ?? '',
      name: json['name'] as String,
      icon: json['icon'] as String?,
      color: json['color'] as String?,
    );
  }

  Map<String, dynamic> toCreateJson() {
    return {
      'name': name,
      if (icon != null) 'icon': icon,
      if (color != null) 'color': color,
    };
  }
}